packages/runtime/container-runtime/src/dataStores.ts

```diff
@@ -493,6 +528,21 @@ export class DataStores implements IDisposable {
         }
         return outboundRoutes;
     }
+
+    /**
+     * This method should be replaced by smearing these map inserts across
+     * all 'add datastore' operations. Its purpose for now is to not create
+     * an alias blob unless we process an alias message.
+     */
+    private buildAliasMap() {
```

Comment:
don't we need to include the mapping in the alias blob here as well if it exists?

Reply:
Can you clarify? I don't believe I follow. This class is getting the alias map from the blob, this method adds to it (if needed). Then when the blob is written, the runtime will read the map from this class.

Reply:
this design is a bit odd to me. the alias blob will duplicate the datastores map for every data store. should we remove all of these before we serialize rather than duplicating

Reply:
Synced offline. This will be done without duplication.